src/condition/lfi_detector.cpp

```diff
@@ -19,123 +17,121 @@ namespace {
-                if (path.find(key) != std::string_view::npos) {
-                    strings.emplace_back(key);
-                }
-            }
-
-            if (object::is_container(&child)) {
-                if (stack.size() < object_limits::default_max_container_depth) {
-                    stack.push(&child);
-                }
-                continue;
-            }
-
-            if (child.type == DDWAF_OBJ_STRING && child.nbEntries >= min_str_len) {
-                const std::string_view value{
-                    child.stringValue, static_cast<std::size_t>(child.nbEntries)};
-                if (path.find(value) != std::string_view::npos) {
-                    strings.emplace_back(value);
-                }
-            }
+    object::kv_iterator it(&params, {}, objects_excluded, limits);
+    for (; it; ++it) {
+        if (deadline.expired()) {
+            throw ddwaf::timeout_exception();
         }
-    }
 
-    return strings;
-}
-
-std::vector<std::string_view> split(std::string_view str, char sep)
-{
-    std::vector<std::string_view> components;
+        const ddwaf_object &param = *(*it);
+        if (param.type != DDWAF_OBJ_STRING || param.stringValue == nullptr ||
+            param.nbEntries < min_str_len) {
+            continue;
+        }
 
-    std::size_t start = 0;
-    while (start < str.size()) {
-        const std::size_t end = str.find(sep, start);
+        std::string_view value{param.stringValue, static_cast<std::size_t>(param.nbEntries)};
+        if (value.find('/') == std::string_view::npos || !path.ends_with(value)) {
+            continue;
+        }
 
-        if (end == start) {
-            // Ignore zero-sized strings
-            start = end + 1;
+        if (value[0] == '/' && value == path) {
+            return {{std::string(value), it.get_current_path()}};
         }
 
-        if (end == std::string_view::npos) {
-            // Last element
-            components.emplace_back(str.substr(start));
-            start = str.size();
-        } else {
-            components.emplace_back(str.substr(start, end - start));
-            start = end + 1;
+        auto parts = split(value, '/');
+        if (parts.size() > 1 && std::find(parts.begin(), parts.end(), "..") != parts.end()) {
+            return {{std::string(value), it.get_current_path()}};
         }
     }
 
-    return components;
+    return {};
 }
 
-std::pair<bool, std::string> lfi_impl(std::string_view path, const ddwaf_object &params)
+lfi_result lfi_impl_windows(std::string_view path, const ddwaf_object &params,
+    const exclusion::object_set_ref &objects_excluded, const object_limits &limits,
+    ddwaf::timer &deadline)
 {
-    auto matched_params = match_params(path, params);
-    if (matched_params.empty()) {
-        return {};
+    auto rpath = path;
+    auto drive_end = path.find(':');
+    if (drive_end != std::string_view::npos) {
+        rpath = path.substr(drive_end + 1, path.size() - (drive_end + 1));
```

Comment:
what would happen if the path ends with `:`?

Reply:
I made some changes, basically if the path ends in `:`, `rpath` will be empty so it's irrelevant for further checks. However if the parameter is the same drive (e.g. `C:`) it'll match.